packages/chakra-components/src/components/Election/Envelope.tsx

```diff
@@ -0,0 +1,117 @@
+import { useClient, useDatesLocale, useElection } from '@vocdoni/react-providers'
+import {
+  ElectionResultsTypeNames,
+  ElectionStatus,
+  IChoice,
+  IQuestion,
+  IVoteEncryptedPackage,
+  IVotePackage,
+  PublishedElection,
+} from '@vocdoni/sdk'
+import { Text } from '@chakra-ui/react'
+import { chakra, ChakraProps, useMultiStyleConfig } from '@chakra-ui/system'
+import { format } from 'date-fns'
+
+export type VotePackageType = IVotePackage | IVoteEncryptedPackage
+
+export const Envelope = ({
+  votePackage,
+  ...props
+}: {
+  votePackage: VotePackageType
+} & ChakraProps) => {
+  const styles = useMultiStyleConfig('Envelope')
+  const { localize } = useClient()
+  const { election } = useElection()
+  const locale = useDatesLocale()
+
+  if (
+    !election ||
+    'encrypted' in votePackage ||
+    !(election instanceof PublishedElection) ||
+    election?.status === ElectionStatus.CANCELED
+  )
+    return null
+
+  if (election?.electionType.secretUntilTheEnd && election.status !== ElectionStatus.RESULTS) {
+    return (
+      <Text sx={styles.secret} {...props}>
+        {localize('results.secret_until_the_end', {
+          endDate: format(election.endDate, localize('results.date_format'), { locale }),
+        })}
+      </Text>
+    )
+  }
+
+  return (
+    <chakra.div sx={styles.wrapper} {...props}>
+      {election.questions.map((q, i) => {
+        return (
+          <chakra.div sx={styles.question}>
+            <Text sx={styles.title}>{localize('envelopes.question_title', { title: q.title.default })}</Text>
+            <ChoosedOptions question={q} questionIndex={i} votes={votePackage.votes} election={election} />
+          </chakra.div>
+        )
+      })}
+    </chakra.div>
+  )
+}
+
+const ChoosedOptions = ({
```

Comment:
Paraphrasing the boss: terrible name. Name it `SelectedOptions` or `EnvelopeOptions` or `EnvelopeChoice`... 

`Choosed` does not exist btw, it's chose, with a single `o`

Reply:
ec2718e

Reply:
ef20ff6